### PR TITLE
Add UBSan kokoro configuration

### PR DESCRIPTION
**Pull request by afd**
Adds a kokoro configuration for UBSan, to allow this sanitizer to be
used during continuous integration.

### CODE REVIEW

File: kokoro/scripts/linux/build-docker.sh [hunk 1]
```diff
@@ -69,6 +69,11 @@ if [ $TOOL = "cmake" ]; then
   if [ $CONFIG = "ASAN" ]; then
     ADDITIONAL_CMAKE_FLAGS="-DSPIRV_USE_SANITIZER=address,bounds,null"
     [ $COMPILER = "clang" ] || { echo "$CONFIG requires clang"; exit 1; }
+  elif [ $CONFIG = "UBSAN" ]; then
+    # UBSan requires RTTI, and by default UBSan does not exit when errors are
+    # encountered - additional compiler options are required to force this.
+    ADDITIONAL_CMAKE_FLAGS="-DSPIRV_USE_SANITIZER=undefined -DENABLE_RTTI=ON -DCMAKE_C_FLAGS=-fno-sanitize-recover -DCMAKE_CXX_FLAGS=-fno-sanitize-recover"
```
Comment (ben-clayton) [inline kokoro/scripts/linux/build-docker.sh:75]: Bit confused what `SPIRV_USE_SANITIZER=undefined` is all about. A comment would be good.

**Review by ben-clayton — APPROVED** [review]
Note, you'll also need some Google-internal configs in order to enable this.

File: kokoro/scripts/linux/build-docker.sh (hunk 1 above)
Comment (afd) [reply kokoro/scripts/linux/build-docker.sh:77]: It's a SPIR-V Tools CMake option for enabling UBSan (it causes -fsanitize=undefined to get used during compilation).

I have added a comment to clarify.

**Comment by afd** [comment]
Thanks for the approval, @ben-clayton - I also just added you as a reviewer to the CL for the internal config stuff.

**Pull request merged**